MHW_ModManager/Classes/Language.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+    public class Language
+    {
+        private Dictionary<string, string> resources;
+
+        private void LoadLanguage(string language = "")
+        {
+            if (string.IsNullOrEmpty(language))
+            {
+                language = System.Threading.Thread.CurrentThread.CurrentUICulture.Name;
+            }
+
+            this.resources = new Dictionary<string, string>();
+            string dir = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, string.Format("ModManagerData\\lang\\{0}", language));
+            if (Directory.Exists(dir))
+            {
+                var jsonFiles = Directory.GetFiles(dir, "*.json", SearchOption.AllDirectories);
+                foreach (string file in jsonFiles)
+                {
+                    LoadFile(file);
+                }
+            }
+        }
+
+        private void LoadFile(string file)
+        {
+            var content = File.ReadAllText(file, Encoding.Default);
+            if (!string.IsNullOrEmpty(content))
+            {
+                var dict = JsonConvert.DeserializeObject<Dictionary<string, string>>(content);
+                foreach (string key in dict.Keys)
+                {
+                    if (!resources.ContainsKey(key))
+                    {
+                        resources.Add(key, dict[key]);
+                    }
+                    else
+                    {
+                        resources[key] = dict[key];
+                    }
+                }
+            }
+        }
+
+        public void InitLanguage(Control control)
+        {
+            LoadLanguage();
+            if (resources == null)
+                return;
+
+            SetControlLanguage(control);
+            foreach (Control ctrl in control.Controls)
+            {
+                InitLanguage(ctrl);
+            }
+            
+            control.ControlAdded += (sender, e) =>
+            {
+                InitLanguage(e.Control);
+            };
+        }
+
+        private void SetControlLanguage(Control control)
+        {
+            if (resources.ContainsKey(control.Name + ".Text"))
+            {
+                control.Text = resources[control.Name + ".Text"];
+            }
+        }
+
+        public string GetLanguageText(string Translate)
+        {
+            LoadLanguage();
+            if (resources == null)
```

Comment:
I think it's not difficult to solve this problem. Give a KEY value to the specified string to accurately describe the meaning of this text. Search for translation based on KEY, so that it will not be affected by the original content modification. I think putting the translated documents outside the program is more convenient for people to provide and modify the translated documents. They don't need to understand the content of the program, they just need to modify the corresponding characters in the external files and store them in the language folder of their own country.

Reply:
In this way, people can add their own language files according to their needs without embedding the translated files into the program. If JSON is not suitable for language files, it can be changed to XML. Of course, what kind of internationalization plan to use is still your choice. I just put forward different points of view.

Reply:
Personally i think i like the idea of the JSON files as they make it easier for others to access and translate if they aren't as knowledgeable with visual studio or any of that stuff. But they should definitely have better keys for the strings. And if we go this route i dont think its necessary to split the files between UI and other texts.

Reply:
agreed with @BoltManGuy.